src/services/contactFormService.ts

```diff
@@ -0,0 +1,97 @@
+import 'firebase/firestore'
+
+export const ContactStatus = {
+    NOT_CALLED: 0,
+    NEED_TO_CALL_AGAIN: 2,
+    CALLED: 4
+}
+
+export type NewContactReq = {
+    firstName: string,
+    lastName: string,
+    gender: string,
+    email: string,
+    phoneNumber: string,
+    graduationSemester: string,
+    graduationYear: string,
+    school: string,
+    permanentAddress: string,
+    city: string,
+    state: string,
+    zipCode: string,
+    housingComplex: string,
+    parentName: string,
+    parentPhone: string,
+    parentEmail: string,
+    interests: string[],
+    status: number,
+    createdAt: Date
+}
+
+export type Contact = {
+    id: string,
+    firstName: string,
+    lastName: string,
+    gender: string,
+    email: string,
+    phoneNumber: string,
+    graduationSemester: string,
+    graduationYear: string,
+    school: string,
+    permanentAddress: string,
+    city: string,
+    state: string,
+    zipCode: string,
+    housingComplex: string,
+    parentName: string,
+    parentPhone: string,
+    parentEmail: string,
+    interests: string[],
+    status: number,
+    createdAt: Date
+}
+
+interface Doc {
+    id: string
```

Comment:
You don’t have to do this but we should make a promise based firebase service sometime soon and extrapolate this into that file. I can do that

Reply:
Not exactly sure what you mean by this? I thought we were using promises in the services?

Reply:
we have a service that uses observables, we should convert it to use promises so we dont have to duplicate code across services like we do here (ex: getAllAndFlatten<>). If you're still confused we can talk about it later